dashboard/modules/job/common.py

```diff
@@ -49,6 +51,7 @@ def is_terminal(self) -> bool:
         return self.value in {"STOPPED", "SUCCEEDED", "FAILED"}
 
 
+# TODO(aguo): Convert to pydantic model
```

Comment:
This would be great. We only used dataclass to avoid adding pydantic as a dependency. Once it's added we should use it for all of these.

Reply:
I'll tackle this as a follow-up since this is a much riskier change because it requires changes to the actual jobs source of truth data storage.